Add Tracer#Enabled, Instrument#Enabled and corresponding SDK behavior (#4063)
open-telemetry/opentelemetry-specification: specification/metrics/sdk.md

first-seen advisory parameters and log an appropriate error as described in
[duplicate instrument registrations](#duplicate-instrument-registration).

### Instrument enabled

**Status**: [Experimental](../document-status.md)

> The instrument [Enabled](./api.md#enabled) operation MUST return `true` if any
> of the following conditions are true, and `false` otherwise:

Reviewer: If I'm reading this correctly, it says `Enabled` MUST return `true` if meter is disabled or all views are dropped. 

maybe

```suggestion
The instrument [Enabled](./api.md#enabled) operation MUST return `true` if none
of the following conditions apply, and `false` otherwise:
```
?

Author: Yeah it looks like I fudged my logic. The instrument is disabled if MeterConfig.disabled=true or all the resolved views use the drop aggregation. Or flipped, the instrument is enabled if MeterConfig.disabled=false AND at least one resolved view isn't using the drop aggregation.

Alternatively to your suggestion I could just flip the true and false in this sentence: 

> The instrument [Enabled](./api.md#enabled) operation MUST return `false` if any of the following conditions are true, and `true` otherwise:

I think that's slightly more readable, but in either case you need to stop and think about it for a minute.

Reviewer: > The instrument [Enabled](https://github.com/open-telemetry/opentelemetry-specification/pull/api.md#enabled) operation MUST return `false` if any of the following conditions are true, and `true` otherwise:

It sounds slightly confusing for the noop case. There is no meter config or view configuration, but instruments should be disabled anyway.

Author: > There is no meter config or view configuration

`MeterConfig.disabled=false` by default ([relevant spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meterconfig)). If an instrument has no matching view, it should use the default aggregation ([relevant spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#measurement-processing)). 

So by default, `Instrument.Enabled` should return true, but I suppose it required the reader to be aware of the default behavior of `MeterConfig` and measurement processing. Do you think it would help if I added a clarifying note stating the what happens by default?

Reviewer: Thanks for the clarification! I'm fine not mentioning the default/noop case, we can always call it out later if necessary.

first-seen advisory parameters and log an appropriate error as described in
[duplicate instrument registrations](#duplicate-instrument-registration).

### Instrument enabled

**Status**: [Experimental](../document-status.md)

The instrument [Enabled](./api.md#enabled) operation MUST return `false` if any
of the following conditions are true, and `true` otherwise:

* The [MeterConfig](#meterconfig) of the `Meter` used to create the instrument
  has parameter `disabled=true`.
* All [resolved views](#measurement-processing) for the instrument are
  configured with the [Drop Aggregation](#drop-aggregation).

Note: If a user makes no configuration changes, `Enabled` returns `true` since by

> default, `MeterConfig.disabled=false` and instruments use the default

Author: ```suggestion
default `MeterConfig.disabled=false` and instruments use the default
```
